ddl/partition.go

```diff
@@ -555,20 +570,8 @@ func truncateTableByReassignPartitionIDs(t *meta.Meta, tblInfo *model.TableInfo)
 		if err != nil {
 			return errors.Trace(err)
 		}
-
-		var newDef model.PartitionDefinition
-		if tblInfo.Partition.Type == model.PartitionTypeHash {
-			newDef = model.PartitionDefinition{
-				ID: pid,
-			}
-		} else if tblInfo.Partition.Type == model.PartitionTypeRange {
-			newDef = model.PartitionDefinition{
-				ID:       pid,
-				Name:     def.Name,
-				LessThan: def.LessThan,
-				Comment:  def.Comment,
-			}
-		}
+		newDef := def
```

Comment:
It's a shadow copy from meta schema, what happens if there is an error during the loop, the referenced memory is partly modified?

Reply:
Modifying `newDef.ID` will not affect the original `def.ID`, since `newDef` and `def` are `model.PartitionDefinition`, not `*model.PartitionDefinition`. 

The only thing `newDef` and `def` will be related are the `LessThan[:]` fields, but we didn't modify it.

Besides, the old code is also a shallow copy.